lib/file_transfer_component/commands/initiate.rb

```diff
@@ -0,0 +1,43 @@
+module FileTransferComponent
+  module Commands
+    class Initiate
+      include Command
+      attr_writer :file_id
+
+      initializer :name, :uri
+
+      dependency :identifier, Identifier::UUID::Random
+
+      def self.build(name, uri, reply_stream_name: nil)
+        instance = new(name, uri)
+        instance.reply_stream_name = reply_stream_name
+        instance.configure
+        instance
+      end
+
+      def self.call(name, uri, reply_stream_name: nil)
+        instance = build(name, uri, reply_stream_name: reply_stream_name)
+        instance.()
+      end
+
+      def file_id
+        @file_id ||= identifier.get
+      end
+
+      def call
+        stream_name = command_stream_name(file_id)
+        write.(command, stream_name, reply_stream_name: reply_stream_name)
```

Comment:
Move above two lines to a method named `write` in the `Command` generalization, and then call `write` from the actuator.